File: tickpulse/backend/src/utils/response.js
// For centralized response handling
module.exports.sendResponse = {
	success: (res, data = {}, code = 200) =>
		res.status(code).json({ success: true, ...data }),

	error: (res, message, code = 500) =>
		res.status(code).json({ success: false, error: message })
};

File: tickpulse/backend-go/internal/utils/response.go
package utils

import "github.com/gin-gonic/gin"

func Success(c *gin.Context, data gin.H, code ...int) {
	status := 200
	if len(code) > 0 {
		status = code[0]
	}
	payload := gin.H{"success": true}
	for k, v := range data {
		payload[k] = v
	}
	c.JSON(status, payload)
}

func Error(c *gin.Context, message string, code int) {
	c.JSON(code, gin.H{"success": false, "error": message})
}
